torch/fx/passes/infra/partitioner.py

```diff
@@ -67,7 +68,7 @@ def propose_partitions(self) -> List[Partition]:
         def maybe_merge_partition(self_id: int, other_id: int):
             # merged_nodes is the union of nodes in two partition to-be-merged
             merged_nodes = copy(partitions_by_id[self_id].nodes)
-            merged_nodes.update(partitions_by_id[other_id].nodes)
+            merged_nodes.extend(n for n in partitions_by_id[other_id].nodes if n not in merged_nodes)
```

Comment:
Because merged_nodes here is a list, the test `if n not in merged_modes` is O(n), making it quadratic.

Reply:
Okay will upload changes.